From 336a8e1a7ecc8f45f1cad920a7ac936eb7cb97c6 Mon Sep 17 00:00:00 2001
From: Sebanc <sebanc.xda@gmail.com>
Date: Wed, 9 Oct 2019 21:49:49 +0200
Subject: [PATCH] Core Microsoft Surface devices: Fix wifi (1/4)

Allowing autosuspend in btusb module appears to impact wifi stability so disable it.
---
 drivers/bluetooth/btusb.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/bluetooth/btusb.c b/drivers/bluetooth/btusb.c
index 40a4f95..66e0c89 100644
--- a/drivers/bluetooth/btusb.c
+++ b/drivers/bluetooth/btusb.c
@@ -29,6 +29,7 @@
 #include <linux/suspend.h>
 #include <linux/gpio/consumer.h>
 #include <asm/unaligned.h>
+#include <linux/surface_devices_dmi.h>
 
 #include <net/bluetooth/bluetooth.h>
 #include <net/bluetooth/hci_core.h>
@@ -40,6 +41,8 @@
 
 #define VERSION "0.8"
 
+static const struct dmi_system_id devices[] = surface_mwifiex_pcie_devices;
+
 static bool disable_scofix;
 static bool force_scofix;
 static bool enable_autosuspend = IS_ENABLED(CONFIG_BT_HCIBTUSB_AUTOSUSPEND);
@@ -3216,7 +3219,7 @@ static int btusb_probe(struct usb_interface *intf,
 	}
 #endif
 
-	if (enable_autosuspend)
+	if (enable_autosuspend && !dmi_check_system(devices))
 		usb_enable_autosuspend(data->udev);
 
 	err = hci_register_dev(hdev);
-- 
2.20.1

